### DISCUSSION TITLE
If users have an installed Xamarin UWP app, can they update to a Maui WinUI 3 app?

### DISCUSSION THREAD

**Discussion by BurkusCat** [dotnet/maui#644] [Q&A]
Imagine an end user has installed an app that was built in Xamarin UWP and have preferences, local databases, etc. If the developer migrates to Maui (and WinUI 3) and releases an update to their app, how does that look?

Is this something that is even possible (inside and outside of the MS Store)?

**Comment by juansturla** [comment]
In any store you can replace your existing app with a new one. Even if they were build with different tools.
But you have to take into account when you choose to migrate your project. 
If you are using Xamarin.Essentials.Preferences (for example), and you upgrade it to MAUI.Essentials.Preferences, I guess that they are maintained if both applications have the same package properties
The same goes for databases. If you maintain the database connection options and structure, you won't have trouble with the migration.
